src/utils.c

```diff
@@ -11,30 +11,40 @@ bool within_int64_repres(double x) {
   return R_FINITE(x) && x <= (double)INT64_MAX && x >= (double)INT64_MIN;
 }
 
-static R_xlen_t firstNonInt(SEXP x) {
+// used to error if not passed type double but this needed extra is.double() calls in calling R code
+// which needed a repeat of the argument. Hence simpler and more robust to return false when not type double.
+bool isRealReallyInt32(SEXP x) {
```

Comment:
Hmm I'm rethinking. That naming implies a symmetry, e.g. suggesting it could accept `integer` inputs also.

WDYT about `isInt32AsReal`? `realFitsInInt32`/`fitsInInt32`? That highlights we are only interested in whether something currently stored as `double` could be stored as `int` instead.

(for the `int` (32) case, the symmetry part is kinda trivial, because `int` is a strict subset of `double`, but the symmetry _does not_ apply to the `int64_t` case, hence why I'm hesitating)